sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -793,6 +794,20 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
       orderedPartitionSpec.put(colName.toLowerCase, partition(colName))
     }
 
+    // Access SQLConf to get 'Dynamic Partition Inserts' parameter specified dynamically
+    // after HiveClient is created
+    val sqlConf = SparkSession.getActiveSession.get.sessionState.conf
+    Seq(
+      "hive.exec.max.dynamic.partitions",
+      "hive.exec.max.dynamic.partitions.pernode",
+      "hive.exec.max.created.files",
+      "hive.error.on.empty.partition"
```

Comment:
We need only 4 among 6 parameters in https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DML#LanguageManualDML-DynamicPartitionInserts .